dali/operators/generic/permute_batch.h

```diff
@@ -29,17 +29,25 @@ class PermuteBatchBase : public Operator<Backend> {
   }
 
   bool SetupImpl(vector<OutputDesc> &outputs, const workspace_t<Backend> &ws) override {
-    auto curr_batch_size = ws.GetInputBatchSize(0);
     outputs.resize(1);
     auto &input = ws.template InputRef<Backend>(0);
     const auto &in_shape = input.shape();
     outputs[0].type = input.type();
 
     if (has_indices_input_) {
-      GetPerSampleArgument<int>(indices_, "indices", this->spec_, ws, curr_batch_size);
+      auto &idx_in = ws.ArgumentInput("indices");
```

Comment:
We don't know the batch size - it should be taken _from_ the argument, not imposed on it.